app/src/test/java/ch/epfl/qedit/backend/MockDBTest.java

```diff
@@ -37,6 +70,125 @@ private void lockWait() {
         }
     }
 
+    @Test
+    public void testGetStringPool_en() {
+        db.getStringPool(
+                "quiz1",
+                "en",
+                new Callback<Response<Map<String, String>>>() {
+                    @Override
+                    public void onReceive(Response<Map<String, String>> data) {
+                        stringPool = data.getData();
+                        error = data.getError();
+                    }
+                });
+        lockWait();
+        assertEquals(Response.NO_ERROR, error);
+        assertEquals(stringPool_en, stringPool);
+    }
+
+    @Test
+    public void testGetStringPool_fr() {
+        db.getStringPool(
+                "quiz1",
+                "fr",
+                new Callback<Response<Map<String, String>>>() {
+                    @Override
+                    public void onReceive(Response<Map<String, String>> data) {
+                        stringPool = data.getData();
+                        error = data.getError();
+                    }
+                });
+        lockWait();
+        assertEquals(Response.NO_ERROR, error);
+        assertEquals(stringPool_fr, stringPool);
+    }
+
+    @Test
+    public void testGetStringPoolError() {
+        db.getStringPool(
+                "error",
+                "en",
+                new Callback<Response<Map<String, String>>>() {
+                    @Override
+                    public void onReceive(Response<Map<String, String>> data) {
+                        stringPool = data.getData();
+                        error = data.getError();
+                    }
+                });
+        lockWait();
+        assertEquals(DatabaseService.WRONG_DOCUMENT, error);
+        assertNull(stringPool);
+    }
+
+    @Test
+    public void testSupportedLanguageError() {
+        db.getSupportedLanguage(
+                "error",
+                new Callback<Response<List<String>>>() {
+                    @Override
+                    public void onReceive(Response<List<String>> data) {
+                        languages = data.getData();
+                        error = data.getError();
+                    }
+                });
+        lockWait();
+        assertEquals(DatabaseService.WRONG_DOCUMENT, error);
+        assertNull(languages);
+    }
+
+    @Test
+    public void testSupportedLanguage() {
+        db.getSupportedLanguage(
+                "quiz1",
+                new Callback<Response<List<String>>>() {
+                    @Override
+                    public void onReceive(Response<List<String>> data) {
+                        languages = data.getData();
+                        error = data.getError();
+                    }
+                });
+        lockWait();
+        assertEquals(Response.NO_ERROR, error);
+        assertTrue(languages.containsAll(Arrays.asList("en", "fr")));
```

Comment:
I think that Harmcrest's `assertThat(languages, containsInAnyOrder(Arrays.asList("en", "fr")));` checks that the list contains exactly the two languages, so you could merge your two lines.

Reply:
Yes i know, i have tried this but i containsInAnyOrder is not recognize and i don't know why